mapconv.go

```diff
@@ -20,18 +21,22 @@ func ToMap(value interface{}, prefix string) (m map[string]string, err error) {
 		switch v.Kind() {
 		case reflect.Bool:
 			m[path] = strconv.FormatBool(v.Bool())
-		case reflect.Float64:
-			m[path] = strconv.FormatFloat(v.Float(), 'f', 0, 64)
-		case reflect.Int:
+		case reflect.Float32, reflect.Float64:
+			m[path] = strings.TrimRight(strings.TrimRight(strconv.FormatFloat(v.Float(), 'f', 6, 64), "0"), ".")
+		case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
 			m[path] = strconv.FormatInt(v.Int(), 10)
+		case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+			m[path] = strconv.FormatUint(v.Uint(), 10)
 		case reflect.String:
 			m[path] = v.String()
-		case reflect.Map, reflect.Slice, reflect.Interface:
+		case reflect.Array, reflect.Map, reflect.Slice, reflect.Interface:
 			var subMap map[string]string
 			subMap, err = ToMap(v.Interface(), path)
 			for key, value := range subMap {
 				m[key] = value
 			}
+		case reflect.Invalid, reflect.Chan, reflect.Func, reflect.UnsafePointer, reflect.Complex64, reflect.Complex128:
```

Comment:
Is this different than the default case in that we are saying we _won't_ support these types? If so, I think that makes sense.